static/packages/autocomplete.js

```diff
@@ -59,26 +59,22 @@ class Address {
         return address;
     }
 
-    extractStreetDetails(self, detailsStr) {
-        let word = "";
+    extractStreetDetails(self, streetDetails) {
         let streetName = "";
         let streetNumber;
+        let i;
 
-        streetDetails = detailsStr;
-
-        for (i = 0; !self.isNumber(word); i++) {
-            word = streetDetails[i];
-            streetName += streetDetails + ' ';
-        }
+        for (i = 0; !self.isNumber(streetDetails[i]); i++)

```

Comment:
What if a number isn't supplied? This would never stop.

Reply:
yeah weird I didn't catch it